src/graph.jl

```diff
@@ -0,0 +1,74 @@
+using JSON
+using SparseArrays
+using LightGraphs
+abstract type AbstractGraph end
+
+struct Graph<:AbstractGraph
```

Comment:
Readability: This struct definition should look more like the `Partition` definition (which is formatted beautifully!) if possible. Too much spacing makes it hard to parse imo.